docs/prover/cli/options.md

```diff
@@ -688,6 +688,54 @@ struct TokenPair {
 We have two contracts `BankToken.sol` and `LoanToken.sol`. We want `tokenA` of the `tokenPair` to be `BankToken`, and `tokenB` to be `LoanToken`. Addresses take up only one slot. We assume `tokenPair` is the first field of Bank (so it starts at slot zero). To do that, we use:
 `certoraRun Bank.sol BankToken.sol LoanToken.sol --verify Bank:Bank.spec --structLink Bank:0=BankToken Bank:1=LoanToken`
 
+(--contract_recursion_limit)=
+### `--contract_recursion_limit`
+
+**What does it do?**
+Contract inlining can cause recursion (see {ref}`--optimistic_contract_recursion`). This
+option sets the contract recursion level, which is the number of recursive calls
+that the Prover will consider when inlining contracts linked using e.g. `--link` or `--struct_link`.
+
+If a counterexample causes a function to be called recursively more than the
+contract recursion limit, it will report an assertion failure (unless
+{ref}`--optimistic_contract_recursion` is set, in which case the counterexample
+will be ignored).
+The default value is zero (i.e. no recursion is allowed).
+
+**When to use it**
+Use this option when there is recursion due to linking calling Solidity
+functions, and this leads to an assertion failure. In this case one can either
```

Comment:
I'm not sure I understand the "calling Solidity functions" attribute.  Maybe it should be

```suggestion
Use this option when after linking the resulting program calls recursive solidity
functions, and this leads to an assertion failure. In this case one can either
```

Reply:
I suppose the assertion failure might be a real one, for example, reachable within the current contract recursion limit.  The message probably tells when this is the case (like it does for summary recursion, right?).